### ISSUE TITLE
Undecipherable error message from `cabal haddock`

### ISSUE THREAD

**Issue by nrnrnr** [tomjaguarpaw/tilapia#108]
I tried to build Haddock documentation for a project I am working on, but I could not make progress:

```
nr@homedog ~/a/sandbox> cabal haddock
cabal: renderBuildTargetProblem: unexpected status (TargetDisabledByUser,Nothing)
CallStack (from HasCallStack):
  error, called at src/Distribution/Client/CmdErrorMessages.hs:356:14 in main:Distribution.Client.CmdErrorMessages
```

I don't know how to interpret the error message.  Haskell experts where I work were also unable to interpret the error message.

I attach a couple of relevant files.
[sandbox.cabal.txt](https://github.com/tomjaguarpaw/tilapia/files/7953368/sandbox.cabal.txt)
[cabal.project.local.txt](https://github.com/tomjaguarpaw/tilapia/files/7953371/cabal.project.local.txt)

**Comment by tomjaguarpaw** [comment]
Have you tried minimising the broken example? That might help narrow down the cause.

**Comment by nrnrnr** [comment]
Where would I start? Start removing .hs files from my cabal file one at a time?

**Comment by tomjaguarpaw** [comment]
Yes, and removing other configuration options from the cabal file.
